File: hw/cañarte/u2/HW17UnitTest/ComplexOperationsPython/ec_edu_espe_complexoperations_controller/person_controller.py
from datetime import datetime, timedelta

class PersonController:
    @staticmethod
    def compute_age_in_days(birth_date):
        today = datetime.now().date()
        days = (today - birth_date).days
        return days

    @staticmethod
    def compute_age_in_years(birth_date):
        today = datetime.now().date()
        years = today.year - birth_date.year - ((today.month, today.day) < (birth_date.month, birth_date.day))
        return years

File: ws/Vaca/u2/WS23UnitTest/ComplexOperations/ComplexOperations/src/main/java/ec/edu/espe/complexoperations/controller/PersonController.java
package ec.edu.espe.complexoperations.controller;

import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

/**
 *
 * @author Luis Vaca, DCCO-ESPE
 */
public class PersonController {

    public static int computeAgeInDays(LocalDate birthDate) {
        LocalDate today = LocalDate.now();
        return (int) ChronoUnit.DAYS.between(birthDate, today);
    }

    public static int computeAgeInYears(LocalDate birthDate) {
        LocalDate today = LocalDate.now();
        return (int) ChronoUnit.YEARS.between(birthDate, today);
    }
}
